### ISSUE TITLE
App freezes few seconds after starting it

### ISSUE THREAD

**Issue by Rik-Kirtaniya** [rampaa/JL#58]
Ever since the last update, there are certain times when the app simply freezes right after launching it, regardless of whether you are just using the overlay window or have opened any of the settings menus. It works for like 5-8 seconds after launch and then completely stops responding. The only way out is to kill the app from task manager, but once this issue happens, subsequent relaunches of the app continue having this same issue, and the only way seems to be restarting windows to make it work again. Another info that I can give is that this happens regardless of whether textractor is running or not.

**Comment by rampaa** [comment]
> Ever since the last update

Which version were you on before updating to the latest release? At first glance there doesn't seem to be any changes from 1.18.5 to 1.18.6 that can cause the issue you have described. Knowing the exact version that introduced this problem would be really helpful to pinpoint the change that caused this problem. I cannot reproduce the issue you've described myself so I'd really appreciate it if you could pinpoint the earliest release that causes the aforementioned issue for you.

> The only way out is to kill the app from task manager

What's the CPU usage of JL like when it becomes unresponsive? Did you ever try waiting for like 5 minutes just to see if JL becomes responsive again? Do you see anything relevant in logs found in ..JL/Logs? What does the Windows Event Viewer show when JL crashes?

FWIW https://github.com/rampaa/JL/commit/1d4daae564f284a52864b36b1124fa23d66fa006 might have solved this problem. But those GC calls were there from the first JL version so not really sure if it's the same problem reoccuring more frequently randomly or something else entirely.

**Comment by rampaa** [comment]
The new version of JL is out. See if it solves the freezing problem for you.

**Comment by Rik-Kirtaniya** [comment]
Apologies for the late reply.
> Which version were you on before updating to the latest release?

I update the releases as soon as I get notified with the popup, so I believe it was the one right before. The freeze issue only started for me like a couple of weeks before, so it was indeed 1.18.6 that I was on when this happened. I tried reinstalling the same version thinking my install might have somehow got corrupted, but the freeze happened again at that time.
> I cannot reproduce the issue you've described myself

I wish I could pinpoint a more specific method, but the issue kept happening sporadically, mostly when I closed one VN, and opened another one, with a single instance of Textractor x86 and JL x64 running in the background. Other than that, there wasn't anything significant that I can think of.
>What's the CPU usage of JL like when it becomes unresponsive?

It was pretty much 0%.
> Did you ever try waiting for like 5 minutes just to see if JL becomes responsive again?

I did, but the freeze seemed indefinite.
> Do you see anything relevant in logs found in ..JL/Logs?

Surprisingly, no log files were created, unless that location is in some other folder besides the install folder.
> What does the Windows Event Viewer show when JL crashes?

JL did not crash, it just froze and became unresponsive, then I had the usual microsoft windows popup saying "JL is not responding".
>The new version of JL is out. See if it solves the freezing problem for you.

Thank you! I'll keep using it and report here if I face that bug again.
